roofit/roofit/src/RooBernstein.cxx

```diff
@@ -102,36 +91,36 @@ Double_t RooBernstein::evaluate() const
   std::tie(xmin, xmax) = _x->getRange(_refRangeName.empty() ? nullptr : _refRangeName.c_str());
   Double_t x = (_x - xmin) / (xmax - xmin); // rescale to [0,1]
   Int_t degree = _coefList.getSize() - 1; // n+1 polys of degree n
-  RooFIter iter = _coefList.fwdIterator();
 
   if(degree == 0) {
 
-    return ((RooAbsReal *)iter.next())->getVal();
+    return static_cast<RooAbsReal *>(_coefList.at(0))->getVal();
```

Comment:
```suggestion
    return static_cast<RooAbsReal &>(_coefList[0]).getVal();
```
the difference between `[i]` and `at(i)` is not only that it returns a reference instead of pointer, but also `[i]` doesn't check if the index is valid, so it has less overhead (at least for a `std::vector` which is the convention we should follow).

In this case, the code before your change had no bounds checking, so you should also not do it and just use `[]`.